### PR TITLE
Rydberg Drive

### PR DESCRIPTION
**Pull request by mudit2812**
Added `rydberg_drive` function to create drive Hamiltonians for neutral atom pulse programming.

Questions/notes for reviewers:
* Currently, the pulses for the `Hardwarehamiltonian` returned by `drive` (#3936) is overwritten by `rydberg_drive`. Should the `HardwarePulse` object creation be removed from `drive` and only be kept in `rydberg_drive`, and eventually `transmon_drive`?

### CODE REVIEW

File: pennylane/pulse/rydberg.py
Comment (lillian542) [inline pennylane/pulse/rydberg.py:126]:
```suggestion
    where :math:`\Omega`, :math:`\delta` and :math:`\phi` correspond to the amplitude, detuning
```

File: pennylane/pulse/rydberg.py
Comment (lillian542) [inline pennylane/pulse/rydberg.py:127]:
```suggestion
    and phase of the laser, :math:`i` corresponds to the wire index, and :math:`\sigma^\alpha` for
```

File: pennylane/pulse/rydberg.py
Comment (lillian542) [inline pennylane/pulse/rydberg.py:138]:
```suggestion
        phase (Union[float, Callable]): float containing the phase (in radians) of the laser field
```

File: pennylane/pulse/rydberg.py
Comment (lillian542) [inline pennylane/pulse/rydberg.py:222]: Haven't thought about this before, but are we supposed to care about custom wire labels? In theory someone could do `rydberg_drive(amp, phase, detuning, wires='a')`, right? And then I guess we still want it cast to a list?

File: pennylane/pulse/rydberg.py
Comment (lillian542) [inline pennylane/pulse/rydberg.py:254]: 100% agree that `HardwarePulse` creation be removed from `drive` and only be kept here, no reason for it to be in `drive`

**Review by lillian542 — COMMENTED** [review]
Looking good! Left a few comments, mostly documentation suggestions

**Review by Qottmann — COMMENTED** [review]
Code-wise looks good so far!
Looking forward to checking the tests and rendered docs :)

File: pennylane/pulse/rydberg.py
Comment (mudit2812) [reply pennylane/pulse/rydberg.py:222]: That's a good point. This should also be updated in @Qottmann 's PR.

File: pennylane/pulse/rydberg.py
Comment (Qottmann) [inline pennylane/pulse/rydberg.py:130]:
```suggestion
    of :math:`\text{µs}`, and the frequency in units of :math:`\text{MHz}`. It is recommended to also follow
```
Alternatively just MHz in text form (then it wouldnt be bold)

File: pennylane/pulse/rydberg.py
Comment (Qottmann) [inline pennylane/pulse/rydberg.py:120]:
```suggestion
    r"""Returns a :class:`ParametrizedHamiltonian` representing the action of a driving laser
    field,
```

File: pennylane/pulse/rydberg.py
Comment (Qottmann) [inline pennylane/pulse/rydberg.py:132]:
```suggestion
    numbers. This driving term can be combined with an interaction term to create a Hamiltonian describing a
```

File: pennylane/pulse/rydberg.py
Comment (Qottmann) [inline pennylane/pulse/rydberg.py:180]:
```suggestion
    The full Hamiltonian evolution and measuring and expectation value can be executed in a ``qnode``.
```

File: pennylane/pulse/rydberg.py
Comment (Qottmann) [inline pennylane/pulse/rydberg.py:197]: That is a bit misleading, it could be understood that the added term consists of multiple local drives, but really it is one drive acting on wires 0 and 1 (so not fully local either)

Maybe make the example so it constructs two terms, each for qubits 0 and 1 separately (or for that manner for all qubits locally!)

File: tests/pulse/test_rydberg.py
Comment (Qottmann) [inline tests/pulse/test_rydberg.py:102]: remove comment

File: pennylane/pulse/rydberg.py
Comment (Qottmann) [inline pennylane/pulse/rydberg.py:125]:
```suggestion
        \frac{1}{2} \Omega(t) \sum_{q \in \text{wires}} (\cos(\phi)\sigma_q^x - \sin(\phi)\sigma_q^y) -
        \delta(t) \sum_{q \in \text{wires}} \sigma_q^z
```
Optional, this is to be consistent with ``drive``

**Review by Qottmann — COMMENTED** [review]
Very nice @mudit2812 some minor comments mostly concerning the docs :)

**Review by lillian542 — APPROVED** [review]
>Currently, the pulses for the Hardwarehamiltonian returned by drive (https://github.com/PennyLaneAI/pennylane/pull/3936) is overwritten by rydberg_drive. Should the HardwarePulse object creation be removed from drive and only be kept in rydberg_drive, and eventually transmon_drive?

I think the decision on the drive PR was that the `HardwarePulse` object on the drive function should be removed in this PR, and I think that's still missing and would be nice to add. Otherwise this looks good to me! 🎉

**Comment by mudit2812** [comment]
>I think the decision on the drive PR was that the HardwarePulse object on the drive function should be removed in this PR, and I think that's still missing and would be nice to add.

It's been removed 😄

**Review by Qottmann — APPROVED** [review]
Looks good to me 👍 I'd wait for the parent PR to be merged before merging this one

File: pennylane/pulse/rydberg.py
Comment (trbromley) [inline pennylane/pulse/rydberg.py:137]:
```suggestion
        phase (Union[float, Callable]): float or callable containing the phase (in radians) of the laser field
```

File: pennylane/pulse/rydberg.py
Comment (trbromley) [inline pennylane/pulse/rydberg.py:144]:
```suggestion
        ParametrizedHamiltonian: a :class:`~.ParametrizedHamiltonian` representing the action of the laser field on the Rydberg atoms.
```

**Review by trbromley — COMMENTED** [review]
Thanks @mudit2812!

**Pull request merged**
